vault/token_store.go

```diff
@@ -1131,16 +1132,9 @@ func (ts *TokenStore) revokeSalted(ctx context.Context, saltedID string) (ret er
 			lock.Lock()
 			defer lock.Unlock()
 
-			// Lookup the token again to make sure something else didn't
-			// revoke in the interim
-			entry, err := ts.lookupSalted(ctx, saltedID, true)
```

Comment:
I believe removing this call introduces a race condition. If another thread removes the token before we grab the write lock above we will restore the deleted entry.

Reply:
Ah, yeah good point.